src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -0,0 +1,36 @@
+import { Request, Response } from "express";
+import axios from "axios";
+import { GitHubServerApp } from "models/github-server-app";
+import { isValidUrl } from "utils/is-valid-url";
+
+export const JiraServerUrlPost = async (
+	req: Request,
+	res: Response
+): Promise<void> => {
+	const { gheServerURL, installationId } = req.body;
+
+	req.log.debug(`Verifying provided GHE server url ${gheServerURL} is a valid URL`);
+	const isGheUrlValid = isValidUrl(gheServerURL);
+
+	if (isGheUrlValid) {
+		try {
+			const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
+
+			if (gitHubServerApps?.length) {
+				req.log.debug(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
+				res.status(200).send({ success: true, moduleKey: "github-list-apps-page" });
+			} else {
+				req.log.debug(`No existing GitHub apps found for url: ${gheServerURL}. Making request to provided url.`);
+				await axios.get(gheServerURL);
+				res.status(200).send({ success: true, moduleKey: "github-app-creation-page" });
+			}
+		} catch (err) {
+			req.log.debug(`Something went wrong: ${gheServerURL}`, err);
+			// TODO - adding error mapping (create confluence page with error codes)
```

Comment:
Already started [page](https://hello.atlassian.net/wiki/spaces/PF/pages/1787434478/Decision+How+do+we+want+to+map+errors+for+failures+on+the+GHE+Server+URL+screen) and wokring on [ARC-1473](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a%2C70121%3A810e719d-1fc7-42f2-9152-27b0f5e44ec8&selectedIssue=ARC-1473)